src/components/TodoElement/TodoElement.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo
+  isLoading: boolean
+  removeTodo: (id: number) => void
+  updateTodo: (id: number, data: Partial<Omit<Todo, 'id' | 'userId'>>) => void
+};
+
+export const TodoElement: React.FC<Props> = ({
+  todo,
+  isLoading,
+  removeTodo,
+  updateTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [titleQuery, setTitleQuery] = useState(todo.title);
+  const { id, title, completed } = todo;
+
+  const cancelEditing = () => {
+    setIsEditing(false);
+  };
+
+  const submitEditing = () => {
+    if (titleQuery === '') {
+      removeTodo(todo.id);
+    }
+
+    if (todo.title !== titleQuery) {
+      updateTodo(id, { title: titleQuery });
+    }
+
+    cancelEditing();
+  };
+
+  const handleEditingForm = (event: React.FormEvent) => {
+    event.preventDefault();
+    submitEditing();
+  };
+
+  useEffect(() => {
+    const handleEscape = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        cancelEditing();
+      }
+    };
+
+    document.addEventListener('keydown', handleEscape);
```

Comment:
i know that with useRef it is ok to refer to the document directly, but you can use onKeyUp attribute on input and it will be work the same... also if you prefer to leave that part of code like that just switch keydown to keyup coz in readme file said that we need to use keyup